src/networking.c

```diff
@@ -1349,13 +1349,43 @@ void acceptCommonHandler(connection *conn, struct ClientFlags flags, char *ip) {
         connClose(conn);
         return;
     }
+    in_addr_t ip_addr = 0;
+    if (sa != NULL && sa->ss_family == AF_INET) ip_addr = ((struct sockaddr_in *)sa)->sin_addr.s_addr;
+
+    if (server.trustedIPCount && ip_addr) {
+        if (/*!isTrustedNetwork(c) &&*/ !checkTrustedIP(ip_addr)) {
```

Comment:
Why `checkTrustedIP` is checked twice both on L1356 and L1388? as the implementation stands now, L1356 supersedes L1388. 

This also brings to my second question on the exact definition of "trusted addresses" - the implementation is saying "untrusted" clients can't access the server. This is not what I had in mind originally and I would agree with @soloestoy that iptables would be the right solution for this particular use case. On the other hand, I was thinking about exempting certain IP addresses from the server policies only (such as `maxclient` on L1388). This way, we only need to pay for the expensive check when the policy is **violated**. In this sense, I think "trusted" is actually a misnomer. Maybe consider `admin_ip` instead?